Modules/HMPID/src/HmpidTask.cxx

```diff
@@ -0,0 +1,247 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   HmpidTask.cxx
+/// \author My Name
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TMath.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "HMPID/HmpidDecodeRawMem.h"
+#include "HMPID/HmpidTask.h"
+
+namespace o2::quality_control_modules::hmpid
+{
+
+HmpidTask::~HmpidTask()
+{
+  if (hPedestalMean) {
+    delete hPedestalMean;
+  }
+
+  if (hPedestalSigma) {
+    delete hPedestalSigma;
+  }
+    
+  if (hBusyTime) {
+    delete hBusyTime;
+  }
+    
+  if (hEventSize) {
+    delete hEventSize;
+  }
+    
+}
+
+Int_t nCycles = 0;
+
+void HmpidTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info) << "initialize HmpidTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
+    ILOG(Info) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+  }
+
+  hPedestalMean = new TH1F("hPedestalMean", "Pedestal Mean", 2000, 0, 2000);
+  hPedestalMean->SetXTitle("Pedestal mean (ADC channel)"); hPedestalMean->SetYTitle("Entries/1 ADC");
+  
+  hPedestalSigma = new TH1F("hPedestalSigma", "Pedestal Sigma", 100, 0, 10);
+  hPedestalSigma->SetXTitle("Pedestal sigma (ADC channel)"); hPedestalMean->SetYTitle("Entries/0.1 ADC");
+  
+  hBusyTime = new TH1F("hBusyTime", "Average Busy Time", 14, 0, 14);
+  hBusyTime->SetXTitle("Equipment"); hBusyTime->SetYTitle("Busy time (#mus)");
+  hBusyTime->SetMarkerStyle(20);
+  
+  hEventSize = new TH1F("hEventSize", "Average Event Size", 14, 0, 14);
+  hEventSize->SetXTitle("Equipment"); hEventSize->SetYTitle("Event size (kB)");
+  hEventSize->SetMarkerStyle(20);  
+
+  getObjectsManager()->startPublishing(hPedestalMean);
+  getObjectsManager()->addMetadata(hPedestalMean->GetName(), "custom", "34");
+  
+  getObjectsManager()->startPublishing(hPedestalSigma);
+  getObjectsManager()->addMetadata(hPedestalSigma->GetName(), "custom", "34");
+
+  getObjectsManager()->startPublishing(hBusyTime);
+  getObjectsManager()->addMetadata(hBusyTime->GetName(), "custom", "34");
+
+  getObjectsManager()->startPublishing(hEventSize);
+  getObjectsManager()->addMetadata(hEventSize->GetName(), "custom", "34");  
+}
+
+void HmpidTask::startOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info) << "startOfActivity" << ENDM;
+  hPedestalMean->Reset();
+  hPedestalSigma->Reset();
+  hBusyTime->Reset();
+  hEventSize->Reset();
+}
+
+void HmpidTask::startOfCycle()
+{
+  ILOG(Info) << "startOfCycle" << ENDM;
+}
+
+void HmpidTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  nCycles++;
+
+  Printf("******************IN HMPID MONITOR DATA*********************************************");
```

Comment:
Same remark about logging. The output will be lost if you print it instead of sending it to either the infologger or fairlogger.